async_dns/base.py

```diff
@@ -109,6 +109,9 @@ def parse(self, data, l):
             elif self.qtype == types.SRV:
                 # priority, weight, port, hostname
                 self.data = struct.unpack('!HHH', data[l: l + 6]) + (utils.load_name(data, l + 6)[1], )
+            elif self.qtype == types.NAPTR:
+                # order, preference, flags, service
+                self.data = struct.unpack('!HHH', data[l: l + 6]) + (utils.load_name(data, l + 6)[1], )
```

Comment:
`flags` and `service` seem to be decoded incorrectly.
refs: <https://tools.ietf.org/html/rfc2915> <https://tools.ietf.org/html/rfc1035>